src/test/java/io/confluent/connect/jdbc/source/integration/JdbcSourceConnectorIT.java

```diff
@@ -0,0 +1,249 @@
+
+    // Add mysql dialect related configurations.
+    addConnectionProperties(props);
+
+    props.put("mode", "incrementing");
+    props.put("incrementing.column.name",  "id");
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expectedNumTasks = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expectedNumTasks);
+
+    log.info("Waiting for records in destination topic ...");
+    ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(
+        NUM_RECORDS,
+        CONSUME_MAX_DURATION_MS,
+        "sql-mysqlTable");
+
+    assertRecordOffset(startIndex, startIndex+NUM_RECORDS, records);
+
+    // delete the connector
+    connect.deleteConnector(CONNECTOR_NAME);
+
+    sendTestDataToMysql(startIndex+NUM_RECORDS, connection);
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expected = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expected);
+
+    records = connect.kafka().consume(
+    NUM_RECORDS*2,
+    CONSUME_MAX_DURATION_MS,
+    "sql-mysqlTable");
+
+    assertRecordOffset(startIndex, (startIndex+NUM_RECORDS*2), records);
+
+  }
+
+  @Test
+  public void testConnectorModeChange() throws InterruptedException, SQLException {
+    int startIndex = 0;
+    dropTableIfExists(KAFKA_TOPIC, connection);
+    createTimestampTable(connection);
+    sendTestTimestampDataToMysql(startIndex, connection);
+
+    connect.kafka().createTopic("sql-" + KAFKA_TOPIC);
+
+    // Add mysql dialect related configurations.
+    addConnectionProperties(props);
+    props.put("mode", "timestamp");
+    props.put("timestamp.column.name",  "time1");
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expectedNumTasks = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expectedNumTasks);
+
+    log.info("Waiting for records in destination topic ...");
+    ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(
+        NUM_RECORDS,
+        5000,
+        "sql-mysqlTable");
+
+    assertRecordOffset(startIndex, NUM_RECORDS, records);
+
+    // delete the connector
+    connect.deleteConnector(CONNECTOR_NAME);
+
+    // change mode
+    props.replace("mode", "incrementing");
+    props.put("incrementing.column.name",  "id");
+    props.remove("timestamp.column.name");
+
+    sendTestTimestampDataToMysql(startIndex+NUM_RECORDS, connection);
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expected = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expected);
+
+    records = connect.kafka().consume(
```

Comment:
maybe add a comment stating that connector will process records from the beginning.